### PR TITLE
Crystal Agitators

### PR DESCRIPTION
**Pull request by Geevies**
Crystal Agitators, large machines that draw power directly from power cables in order to agitate asteroid turfs, causing them to form grindable chemical crystals, which can be mined with standard mining equipment, or bashed with any sort of heavy duty weaponry.

### CODE REVIEW

**Comment by Geevies** [comment]
I'm waiting for sprites that aren't my codersprites, but I'm going to open this for !review so you guys can check it out for errors so long. Any developers who review this should probably ping me in #developers for details.

File: code/game/objects/structures/phoron_crystals.dm [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	name = "phoron crystal cluster"
+	desc = "A cluster of hardened phoron crystals."
+	icon = 'icons/obj/phoron_crystals.dmi'
+	icon_state = "phoron_crystals"
+	anchored = TRUE
+	density = FALSE
+	layer = ABOVE_CABLE_LAYER
+	var/state = 0
+	var/health = 100
+	var/cover = 20 //how much cover the crystals provides against projectiles.
```
Comment (Geevies) [inline code/game/objects/structures/phoron_crystals.dm:11]:
```suggestion
```

File: code/game/objects/structures/phoron_crystals.dm [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	P.health *= health_mod
+	qdel(src)
+
+/obj/structure/phoron_crystal/dense
+	name = "dense phoron crystal cluster"
+	desc = "A dense cluster of hardened phoron crystals."
+	icon_state = "dense_phoron_crystals"
+	density = TRUE
+	health = 200
+	cover = 60
```
Comment (Geevies) [inline code/game/objects/structures/phoron_crystals.dm:126]:
```suggestion
```

**Comment by NonQueueingMatt** [comment]
reopening when not stale

File: code/modules/mob/living/carbon/human/human_species.dm [hunk 3]
```diff
@@ -192,6 +192,56 @@ INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy/mannequin)
...
+	name = real_name
+	dna.real_name = real_name
+	if(mind)
+		mind.name = real_name
+	status_flags |= NO_ANTAG
+	disabilities |= PACIFIST
+
+	equip_to_slot_or_del(new /obj/item/device/radio/headset/headset_penal(src), slot_l_ear)
+
+	remote_network = "prisonrobots"
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/human_species.dm:208]: These should probably be defines or datums at this point

File: code/modules/power/crystal_agitator.dm [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			continue
+		if(locate(/obj/structure/reagent_crystal/dense) in T)
+			continue
+		grow_turfs += T
+
+	var/turf/selected_turf = pick(grow_turfs)
+	if(!selected_turf || !istype(selected_turf))
+		return
+	if(locate(/obj/structure/reagent_crystal) in selected_turf)
+		var/obj/structure/reagent_crystal/P = locate() in selected_turf
```
Comment (NonQueueingMatt) [inline code/modules/power/crystal_agitator.dm:65]: You can just do locate() once and then check if that var exists

File: code/modules/recycling/disposal.dm [hunk 5]
```diff
@@ -104,6 +108,14 @@
 		update()
 		return
 
+	else if(istype(I, /obj/item/storage/bag/crystal) || istype(I, /obj/item/storage/bag/ore))
+		var/obj/item/storage/bag/ore/B = I
+		to_chat(user, SPAN_NOTICE("You empty \the [B] into \the [src]."))
+		for(var/obj/item/O in B.contents)
+			B.remove_from_storage(O, src)
+		update()
+		return
```
Comment (NonQueueingMatt) [inline code/modules/recycling/disposal.dm:117]: Shouldn't this be for every storage/bag item at this point?

Comment (Geevies) [reply code/modules/recycling/disposal.dm:117]: ehhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh nah i don't think so. there are valid reasons for wanting to dump the entire bag. i could make it an intent check but that seems very clunky

Comment (NonQueueingMatt) [reply code/modules/recycling/disposal.dm:117]: How is it clunky, just check for like harm intent and make all storage items drop all contents into the disposal unit, far cleaner and makes more sense

**Comment by Geevies** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

**Review by skull132 — APPROVED** [review]

**Pull request merged**